### PR TITLE
update project to use JDK 21

### PR DESCRIPTION
**Pull request by jeffreyscarpenter**
Update project from JDK 17 to JDK 21

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/docker/Dockerfile.jvm [hunk 1]
```diff
@@ -77,7 +77,7 @@
 #   accessed directly. (example: "foo.example.com,bar.example.com")
 #
 ###
-FROM registry.access.redhat.com/ubi8/openjdk-17-runtime:1.20
+FROM registry.access.redhat.com/ubi8/openjdk-21-runtime:1.20
```
Comment (tatu-at-datastax) [inline src/main/docker/Dockerfile.jvm:80]: We have 4 Dockerfiles, should probably update others. Or at least `Dockerfile-profiling.jvm`?

**Review by tatu-at-datastax — COMMENTED** [review]
Looks good but missing change to `Dockerfile-profiling.jvm`.

File: src/main/docker/Dockerfile.jvm (hunk 1 above)
Comment (jeffreyscarpenter) [reply src/main/docker/Dockerfile.jvm:80]: yes, I missed that one. Fixed

File: src/main/docker/Dockerfile-profiling.jvm [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # based on less minimal UBI8 OpenJDK 17 image
 #
 ###
-FROM registry.access.redhat.com/ubi8/openjdk-17:1.20
+FROM registry.access.redhat.com/ubi8/openjdk-21:1.20
```
Comment (tatu-at-datastax) [inline src/main/docker/Dockerfile-profiling.jvm:6]: I assume you checked but is `1.20` the latest available minor version?

Comment (jeffreyscarpenter) [reply src/main/docker/Dockerfile-profiling.jvm:6]: > is 1.20 the latest available minor version?

Yes https://catalog.redhat.com/software/containers/ubi8/openjdk-21-runtime/653fd184292263c0a2f14d69?gs&q=openjdk%2021%20ubi

**Pull request merged**